platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/result/DefaultResolvedArtifactResult.java

```diff
@@ -65,7 +81,17 @@ public File getFile() {
     }
 
     @Override
+    @Deprecated
     public ResolvedVariantResult getVariant() {
         return variant;
     }
+
+    @Override
+    public Set<ResolvedVariantResult> getVariants() {
+        return variants;
+    }
+
+    public DefaultResolvedArtifactResult withAddedVariant(AttributeContainer variantAttributes, List<? extends Capability> capabilities, DisplayName variantName) {
+        return new DefaultResolvedArtifactResult(this, new DefaultResolvedVariantResult(identifier.getComponentIdentifier(), variantName, variantAttributes, capabilities, null));
```

Comment:
❓ Is the last argument to `DefaultResolvedVariantResult` always `null`?  Can it be removed?